Initial support for deployment variables

sl4mmy opened the pull request.

sl4mmy commented:
This PR depends on #120, hence why it appears there is a duplicate commit (7f6b700) between these two PRs.  Please merge  #120 first as a fast-forward only commit to avoid any merge conflicts or duplicate commits.

ktrysmt reviewed (commented):
Looks good, but leave a comment.

sl4mmy commented on repository.go:
```diff
@@ -426,15 +467,115 @@ func (r *Repository) BranchingModel(rbmo *RepositoryBranchingModelOptions) (*Bra
+	if opt.MaxDepth > 0 {
+		params.Add("max_depth", strconv.Itoa(opt.MaxDepth))
+	}
+ */
```
Yeah, I will open a support ticket with Atlassian to enquire about it.  It *seems* like an oversight, imo, but if they say it was intentionally not added then I will happily create a new PR to remove the entire commented block.

sl4mmy replied:
I'm discovering lots of little inconsistencies like this in BitBucket's API.  This one was just more glaring because I originally copied & pasted the code from `repository.ListPipelineVariables` and then spent a while scratching my head trying to figure out why this was failing.  Took a bit of trial & error, and wild guessing, on my part before I tried removing the request params and it started working.  🤦

ktrysmt replied:
Exactly. BitBucket API v2 had been sometime crush or behave strangely. As far as I know, time will solve that.

sl4mmy replied:
Ok, the commented-out code has been re-enabled.

ktrysmt commented:
I want to wait atlassian's activity for a while.

Then, Please rebase this PR with origin/master. via https://github.com/ktrysmt/go-bitbucket/pull/120. @sl4mmy

sl4mmy commented:
> I want to wait atlassian's activity for a while.
> 
> Then, please rebase this PR with origin/master. via #120. @sl4mmy

Sounds good, will do.  (Already went ahead and rebased this on top of master to avoid any further confusion)

sl4mmy commented:
After conferring with Atlassian this should be working now.  I confirmed that paging the list of deployment variables works as expected; I removed the comment blocks and updated the test to pass `MaxDepth`, `PageNum`, and `Pagelen` as request params.

ktrysmt commented:
It's a good news for us. Thank you for your report. @sl4mmy

ktrysmt reviewed (approved):
LGTM. 👍
